packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -368,14 +554,38 @@ export class InternalModelManager {
     });
   }
 
+  /**
+   * Deletes the records in the backend datastore that match the given filter criteria. Uses the Internal API.
+   *
+   * Does *not* run actions -- use the Public API for that.
+   *
+   * @example
+   * // removes all unpublished posts from the database
+   * await api.internal.post.deleteMany({filter: { published: { equals: false } } });
+   *
+   * @param options Search and filter options for the records to delete
+   */
   async deleteMany(options?: { search?: string; filter?: RecordData }): Promise<void> {
     return await this.transaction(async (transaction) => {
       const response = await transaction.client.mutation(internalDeleteManyMutation(this.apiIdentifier), options).toPromise();
       assertMutationSuccess(response, ["internal", `deleteMany${this.capitalizedApiIdentifier}`]);
     });
   }
 
+  /** @private */
   async transaction<T>(callback: (transaction: GadgetTransaction) => Promise<T>): Promise<T> {
     return await this.connection.transaction(callback);
   }
 }
+
+function formatInternalSelectVariable(select: InternalFieldSelection | undefined): undefined | string[] {
```

Comment:
If we pass `select: { id: false }` that's the equivalent of passing `[]`.

I'm not sure why we have support for `Record<string, boolean>` instead of just a string array. I know it's to match our existing API but what does this object syntax get you that the array doesn't?

Reply:
Good question! It gets consistency with all the other api functions in the JS client. The Public API supports the nested selection style, and I think that'll be what acquaints most folks with the idea of JS selections at all. I think its kind of regrettable that the internal api works different at a graphql level, but since just about everyone is actually using the JS client and doesn't really care what it does under the hood, I think we should make it quack as similarly as possible!